aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -939,6 +939,34 @@ Tensor& add_out_sparse_csr_cpu(
       return out;
     }
 
+    // TODO: insert a shape-based switch
+    const auto self_indices_dtype = std::get<0>(at::sparse_csr::getCompressedPlainIndices(self)).scalar_type();
+    const auto other_indices_dtype = std::get<0>(at::sparse_csr::getCompressedPlainIndices(other)).scalar_type();
+    // TODO: prevent int32 out indices with int64 inputs
+
+    if (self_indices_dtype == at::kLong || other_indices_dtype == at::kLong) {
+      auto res = other.to_sparse().add(self.to_sparse(), alpha).to_sparse(out.layout());
+      res = self.to_dense().add(other.to_dense(), alpha).to_sparse(out.layout());
+
```

Comment:
OK, looks like `add` produces inputs which are marked as coalesced while not being such.